### PR TITLE
helm: Include seLinuxMount only if KubeVersion greater or equal of 1.25

### LINKED ISSUE #4437: Make seLinuxMount optional in CSIDriver for compatibility
# Describe the bug #

According to https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/ the seLinuxMount field in the CSIDriver object is introduced as alpha in K8S 1.25, made beta in 1.27 and only starting from 1.28 defaults to true.
If this feature gate is not enabled, ceph-csi-rbd and ceph-csi-fs fail when installing.
I would like for seLinuxMount to be optional but true by default, to have better compatibility 

This was enabled on tag 3.10.2. Previous tags did not have this issue. See PR https://github.com/ceph/ceph-csi/pull/4346

# Environment details #

- Image/version of Ceph CSI driver : 
- Helm chart version : v3.10.2
- Kernel version : 
- Mounter used for mounting PVC (for cephFS its `fuse` or `kernel`. for rbd its
  `krbd` or `rbd-nbd`) : krbd
- Kubernetes cluster version : v1.25.6
- Ceph cluster version : 

# Steps to reproduce #

1. Deploy ceph-csi-rbd on a cluster without the feature flag SELinuxMountReadWriteOncePod enabled
1. Chart will fail while deploying due to an error: `unknown field "seLinuxMount" in io.k8s.api.storage.v1.CSIDriverSpec`

# Actual results #

Chart failed to install

# Expected behavior #

Chart installs correctly or shows a descriptive error about the feature flag

### LINKED ISSUE #4474: CSI helm charts doesn't verify support of seLinuxMount by k8s
# Describe the bug #

Starting from 3.10.2 chart version seLinuxMount was added without checking k8s cluster version.

# Environment details #

- Image/version of Ceph CSI driver : doesn't matter
- Helm chart version : `3.10.2`
- Kernel version : doesn't matter
- Mounter used for mounting PVC (for cephFS its `fuse` or `kernel`. for rbd its
  `krbd` or `rbd-nbd`) : doesn't matter
- Kubernetes cluster version : 1.24.x and lower
- Ceph cluster version : doesn't matter

# Steps to reproduce #

1. have k8s with version of 1.24.x or lower
2. try install ceph csi helm chart with version 3.10.2
3. see error that `seLinuxMount` not supported but tried to be created by `csidriver-crd.yaml`

# Actual results #

Can't install ceph csi on k8s version lower then 1.25.x

# Expected behavior #

No error occurred

# Logs #

N/A

# Additional context #

[Kubernetes CSI Developer Documentation](https://kubernetes-csi.github.io/docs/csi-driver-object.html#:~:text=seLinuxMount%3A%20true%20%23%20Added%20in%20Kubernetest%201.25)

### PR DESCRIPTION
**Pull request by dragoangel**
# Describe what this PR does #

Fix chart to be compatible with kube version lower then 1.25.0

## Is there anything that requires special attention ##

Do you have any questions?
Yes, I don't see where to bounce helm chart version, `version: 3-canary` looks not legit

Is the change backward compatible?
Yes

Are there concerns around backward compatibility?
No

Provide any external context for the change, if any.
Desribed in issue #4474

## Related issues ##

Mention any github issues relevant to this PR. Adding below line
will help to auto close the issue once the PR is merged.

Fixes: #4474 
Fixes: #4437

## Future concerns ##

List items that are not part of the PR and do not impact it's
functionality, but are work items that can be taken up subsequently.

**Checklist:**

* [ ] **Commit Message Formatting**: Commit titles and messages follow
  guidelines in the [developer
  guide](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#commit-messages).
* [ ] Reviewed the developer guide on [Submitting a Pull
  Request](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#development-workflow)
* [ ] [Pending release
  notes](https://github.com/ceph/ceph-csi/blob/devel/PendingReleaseNotes.md)
  updated with breaking and/or notable changes for the next major release.
* [ ] Documentation has been updated, if necessary.
* [ ] Unit tests have been added, if necessary.
* [ ] Integration tests have been added, if necessary.

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

* `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)

</details>

### CODE REVIEW

**Comment by nixpanic** [comment]
>  Yes, I don't see where to bounce helm chart version, version: 3-canary looks not legit

The version gets updated to something real when a release is made.

**Comment by dragoangel** [comment]
> > Yes, I don't see where to bounce helm chart version, version: 3-canary looks not legit
> 
> The version gets updated to something real when a release is made.

Hi @nixpanic , I also get such feeling by checking git history, but was not sure to 100%, thanks

**Comment by nixpanic** [comment]
@dragoangel , it seems the subject of the commit it a little too long. **commitlint** complains about it:

✖   header must not be longer than 72 characters, current length is 73 [header-max-length]

**Comment by dragoangel** [comment]
> @dragoangel , it seems the subject of the commit it a little too long. **commitlint** complains about it:
> 
> ✖ header must not be longer than 72 characters, current length is 73 [header-max-length]

Will fix, thanks

File: charts/ceph-csi-cephfs/templates/csidriver-crd.yaml [hunk 1]
```diff
@@ -6,4 +6,6 @@ spec:
   attachRequired: false
   podInfoOnMount: false
   fsGroupPolicy: {{ default "File" .Values.CSIDriver.fsGroupPolicy }}
+{{- if semverCompare ">=1.25-0" .Capabilities.KubeVersion.GitVersion }}
```
Comment (iPraveenParihar) [inline charts/ceph-csi-cephfs/templates/csidriver-crd.yaml]: `.Capabilities.KubeVersion.GitVersion` seems to be depcreated in Helm v3?
Can we use `Capabilities.KubeVersion.Version`?

ref - https://helm.sh/docs/chart_template_guide/builtin_objects/

Comment (dragoangel) [reply charts/ceph-csi-cephfs/templates/csidriver-crd.yaml]: Hi @iPraveenParihar I tested that if it works with `GitVersion`, and saw it in many helm charts, but yeah, I can change this, np.

**Comment by dragoangel** [comment]
@iPraveenParihar @Rakshith-R can you please review this PR?

**Review by iPraveenParihar — APPROVED** [review]

**Review by Rakshith-R — APPROVED** [review]
Thanks

**Comment by nixpanic** [comment]
@mergifyio rebase

**Comment by nixpanic** [comment]
@mergifyio queue

**Comment by dragoangel** [comment]
> > queue
> 
> #### 🛑 The pull request has been removed from the queue `default`
> 
> <details>
> 
> The queue conditions cannot be satisfied due to failing checks.
> 
> You can take a look at `Queue: Embarked in merge queue` check runs for more details.
> 
> In case of a failure due to a flaky test, you should first retrigger the CI.
> Then, re-embark the pull request into the merge queue by posting the comment
> `@mergifyio refresh` on the pull request.
> 
> 
> </details>
> 
> 
> 

@nixpanic looks not okay?

**Comment by nixpanic** [comment]
@mergifyio requeue

**Comment by nixpanic** [comment]
> @nixpanic looks not okay?

Sometimes the CI environment fails to pull container images. Hopefully a retest of the failed jobs works now.

**Comment by nixpanic** [comment]
@mergifyio queue

**Comment by nixpanic** [comment]
@mergifyio requeue

**Comment by nixpanic** [comment]
@mergifyio refresh

**Comment by nixpanic** [comment]
@mergifyio rebase

**Comment by nixpanic** [comment]
@mergifyio queue

:crossed_fingers:

**Comment by dragoangel** [comment]
@nixpanic , multi-arch-build build failed 😓

**Pull request merged**